cmd/config-migrate.go

```diff
@@ -2742,3 +2746,45 @@ func migrateV32ToV33MinioSys(objAPI ObjectLayer) error {
 	logger.Info(configMigrateMSGTemplate, configFile, "32", "33")
 	return nil
 }
+
+func migrateV33ToV34MinioSys(objAPI ObjectLayer) error {
+	configFile := path.Join(minioConfigPrefix, minioConfigFile)
+
+	ok, data, err := checkConfigVersion(objAPI, configFile, "33")
+	if err == errConfigNotFound {
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("Unable to load config file. %v", err)
+	}
+	if !ok {
+		return nil
+	}
+
+	cfg33 := &serverConfigV33{}
+	if err = json.Unmarshal(data, cfg33); err != nil {
+		return err
+	}
+	if bool(cfg33.Worm) {
+		logger.Fatal(uiErrInvalidWormValue(nil).Msg("Worm is unsupported"),
```

Comment:
After upgrading the binary if we fail in this manner, it will be super unexpected. Since they have a server down and they can't fix the config because the config is erasure-coded in the backend.

I would suggest a more of a simpler approach, i.e throw a warning and proceed.

Reply:
Also, I see that we may fail twice, first while MINIO_WORM is set in env. and also during when the config.json has persisted in the backend.  What would be preferred is perhaps we throw warning and proceed.

Reply:
This is done as per recommendation of @abperiasamy.

Reply:
This document doesn't say worm is being deprecated for a newer and better approach. It also doesn't state that it should be done before upgrade..

Reply:
He may not be aware of all the nuances that this needs to address, during upgrades. For example if users upgrade using `mc admin update` all servers will fail to start. We need to show users the document on how to turn it off..

Reply:
Probably you could check with @abperiasamy on this issue to do things right.